libs/wingsdk/src/std/json.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Immutable Json
+ */
+export class Json {
+  /**
+   * Returns a specified element from the Json.
+   *
+   * @macro ($self$)[$args$]
+   *
+   * @param key The key of the element to return
+   * @returns The element associated with the specified key, or undefined if the key can't be found
+   */
+  public get(key: string): Json {
+    key;
+    throw new Error("Macro");
+  }
+
+  /**
+   * Returns a specified element at a given index from Json Array
+   *
+   * @macro ($self$)[$args$]
+   *
+   * @param index The index of the element in the Json Array to return
+   * @returns The element at given index in Json Array, or undefined if index is not valid
+   */
+  public getAt(index: number): Json {
```

Comment:
So currently every Json object has the `get_at()` method even if it's not a json array?  Let's see if we can come up with a more typesafe API (no suggestion at the moment - only complains :) ).

Reply:
Yea I feel you here, actually if it wasnt for the fact that `get()` only accepts a string that method could have handled arrays as well 😅 I dont like this DX but I think we will solve this with `dot` and `[]` notation

Reply:
This is fine. `Json` is by definition not type-safe. If you know the structure, you can use `from_json` to turn it into a type-safe object.